src/controller.cpp

```diff
@@ -588,6 +588,7 @@ void Controller::serverUnavailable() {
 
   if ((m_state == StateOn) || (m_state == StateSwitching) ||
       (m_state == StateConnecting)) {
+    NotificationHandler::instance()->serverUnavailableNotification();
```

Comment:
This should probably be done in the `MozillaVPN` class where we connect to the `readyToServerUnavailable` signal to keep the UI behavior separated from the controller. See the `connect()` call at src/mozillavpn.cpp:130